components/script/dom/element.rs

```diff
@@ -692,6 +693,28 @@ impl LayoutElementHelpers for LayoutJS<Element> {
         }
     }
 
+    #[allow(unsafe_code)]
+    // https://html.spec.whatwg.org/multipage/#language
+    fn language(&self) -> Atom {
+        unsafe {
+            let elem = self.unsafe_get();
+            if let Some(attr) = (*elem).get_attr_val_for_layout(&ns!(xml), &local_name!("lang")) {
+                Atom::from(attr)
+            } else if let Some(attr) = (*elem).get_attr_val_for_layout(&ns!(), &local_name!("lang")) {
+                Atom::from(attr)
+            } else {
+                self.upcast::<Node>().parent_node_ref()
```

Comment:
I'd avoid to do this recursively TBH, could we use a iterator/loop instead? It seems that otherwise creating a very deep dom could overflow the stack fairly easily.